#13797493: Текущие заявки в ЛК

djpont opened the pull request:
1. Task: Текущие заявки в ЛК

2. Screenshoot:
![image](https://github.com/agricultural-machinery-rental/client/assets/34692754/32bec00a-a000-4e20-9635-6bb28be3b261)

Catalinadevolg committed:
docs: add architecture doc file

Catalinadevolg committed:
docs: update architecture.md

Catalinadevolg committed:
docs: add readme in docs

djpont committed:
feat: компонент BookingItem

djpont committed:
Merge remote-tracking branch 'origin/dev' into feat/040-booking-current

# Conflicts:
#	src/app/profile/orders/page.tsx

djpont committed:
Merge remote-tracking branch 'origin/documentation' into feat/040-booking-current

# Conflicts:
#	docs/architecture.md

djpont committed:
Merge remote-tracking branch 'origin/dev' into feat/040-booking-current

djpont committed:
feat: поправил ошибки после мерджа dev

djpont committed:
refactor: вынес иконку edit в shared

djpont committed:
refactor: вынес иконку edit в shared

djpont committed:
feat: поправил баг в Breadcrumbs

djpont committed:
feat: собрал в единый CurrentBookingsList

djpont committed:
feat: добавил белый фон

mrHoft reviewed (approved):
Грамотно выполнено.

djpont committed:
Merge branch 'dev' into feat/040-booking-current

Catalinadevolg commented on src/features/Breadcrumbs/ui.tsx:
```diff
@@ -11,21 +11,24 @@ export const Breadcrumbs = () => {
+  return (
+    <StBreadcrumbsItemWrap>
+      {activeBreadcrumbs.map((breadcrumb, index) => (
+        <StBreadcrumbsItem key={breadcrumb.url}>
```
Замени здесь, пожалуйста, а то у нас url не уникальны
```suggestion
        <StBreadcrumbsItem key={index}>
```

Catalinadevolg commented on src/shared/ui/BookingItem/ui.tsx:
```diff
@@ -0,0 +1,33 @@
+        {dates && <StDates>{dates}</StDates>}
+        {category && <StCategory>{category}</StCategory>}
+        {comment && <StComment>{comment}</StComment>}
+        <StEditButton>{button}</StEditButton>
```
`StActionButton` или просто `StButton`

Catalinadevolg commented on src/entities/booking/BookingCurrentItem/typing.ts:
```diff
@@ -0,0 +1,6 @@
+import { type TBookingItem } from '@/shared/ui/BookingItem';
+
+export type TBookingCurrentItem = Pick<TBookingItem, 'photo' | 'name'> & {
+  dates: string;
+  comment: string;
+};
```
А если так?
```suggestion
export type TBookingCurrentItem = Required<Omit<TBookingItem, 'category' | 'button'>>
```
Чтобы не прописывать dates и comment повторно, а исключить ненужное из типа базового компонента и оставшиеся свойства сделать обязательными

djpont replied:
Отлично, так сделаю. Не знал про Required<>. Но я хотел бы оставить Pick, чтобы в случае появления новых полей не делать лишних действий.

Catalinadevolg reviewed (approved):
Отлично сделано!
Мне нравится, как ты подходишь к типизации. Добавила лишь своё небольшое предложение (на твоё усмотрение).

djpont committed:
Merge remote-tracking branch 'origin/dev' into feat/040-booking-current

# Conflicts:
#	src/shared/modal/modalWindows.tsx

djpont committed:
feat: поправил по комментариям в PR

djpont committed:
#13797543: Избранное в ЛК (#64)

1. Task: Избранное в ЛК

2. Screenshot:

![image](https://github.com/agricultural-machinery-rental/client/assets/34692754/0044fb4b-ad6d-4cb8-9a2b-903f7e2b8351)

djpont committed:
Merge branch 'dev' into feat/040-booking-current
